useAppConfig lose types from nuxt schema with new tsconfig

### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v24.2.0
- Nuxt Version:     4.0.1
- CLI Version:      3.26.4
- Nitro Version:    2.12.4
- Package Manager:  pnpm@10.13.1
- Builder:          -
- User Config:      compatibilityDate, experimental, modules, unocss, vue, app, colorMode, css, content, eslint, devtools, i18n, hub, nitro
- Runtime Modules:  @nuxthub/core@0.9.0, @nuxt/eslint@1.7.1, @nuxt/content@3.6.3, @unocss/nuxt@66.3.3, @nuxt/image@1.10.0, @pinia/nuxt@0.11.2, @nuxt/fonts@0.11.4, @nuxtjs/i18n@9.5.6, @nuxtjs/color-mode@3.5.2
- Build Modules:    -
------------------------------

### Reproduction

I tried to reproduce on Stackblitz but it don't seems to be reproductible as I have on my machine.

https://stackblitz.com/edit/github-c4fvnsee?file=tsconfig.json

### Describe the bug

I'm using Nuxt with Nuxt Content.

I migrate to Nuxt 4 and when I change the tsconfig.json to match the new one, typings from `useAppConfig` is losing those coming from `NuxtCustomSchema` (which are correctly generated).  On hovering it shows AppConfig and that's it.

<img width="604" height="244" alt="Image" src="https://github.com/user-attachments/assets/f1429dd4-46fa-44b3-a2e9-d5c9f497cfa9" />

Where I use config then, it shows me typing error.
When I switch to the legacy tsconfig, typing issue are gone.

### Additional context

_No response_

### Logs

```shell-script
Typecheck summary: 



app/layouts/default.vue:35:33 - error TS2339: Property 'title' does not exist on type '{}'.

35   title: appConfig.information?.title,
                                   ~~~~~

app/layouts/default.vue:36:39 - error TS2339: Property 'description' does not exist on type '{}'.

36   description: appConfig.information?.description,
                                         ~~~~~~~~~~~

app/layouts/default.vue:39:36 - error TS2339: Property 'socials' does not exist on type '{}'.

39     github: appConfig.information?.socials?.github,
                                      ~~~~~~~

app/layouts/default.vue:40:38 - error TS2339: Property 'socials' does not exist on type '{}'.

40     linkedin: appConfig.information?.socials?.linkedin,
                                        ~~~~~~~

app/layouts/default.vue:41:40 - error TS2339: Property 'birthdate' does not exist on type '{}'.

41     isBirthday: appConfig.information?.birthdate ? isBirthday(appConfig.information?.birthdate) : false,
                                          ~~~~~~~~~

app/layouts/default.vue:41:86 - error TS2339: Property 'birthdate' does not exist on type '{}'.

41     isBirthday: appConfig.information?.birthdate ? isBirthday(appConfig.information?.birthdate) : false,
                                                                                        ~~~~~~~~~

app/layouts/default.vue:44:27 - error TS2339: Property 'icons' does not exist on type '{}'.

44     normal: appConfig.ui?.icons?.normal as string[] ?? [],
                             ~~~~~

app/layouts/default.vue:45:29 - error TS2339: Property 'icons' does not exist on type '{}'.

45     birthday: appConfig.ui?.icons?.birthday as string[] ?? [],
                               ~~~~~

app/middleware/title.global.ts:5:28 - error TS2571: Object is of type 'unknown'.

5     const appConfigTitle = useAppConfig().information.title;
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~

app/pages/index.vue:14:62 - error TS2339: Property 'messages' does not exist on type '{}'.

14 const messages = useState('randomIndex', () => appConfig.ui?.messages?.sort(() => Math.random() - 0.5))
                                                                ~~~~~~~~

nuxt.schema.ts:3:16 - error TS2304: Cannot find name 'defineNuxtSchema'.

3 export default defineNuxtSchema({
                 ~~~~~~~~~~~~~~~~

nuxt.schema.ts:3:16 - error TS2304: Cannot find name 'defineNuxtSchema'.

3 export default defineNuxtSchema({
                 ~~~~~~~~~~~~~~~~

nuxt.schema.ts:3:16 - error TS2304: Cannot find name 'defineNuxtSchema'.

3 export default defineNuxtSchema({
                 ~~~~~~~~~~~~~~~~
```

@danielroe not sure if related, but here is a reproduction where augmenting `AppConfigInput` doesn't work: https://stackblitz.com/edit/github-nhusdfi6?file=app%2Ftypes%2Findex.d.ts

If you run `nuxt typecheck`, it outputs:
```
app/app.config.ts:2:16 - error TS7031: Binding element 'property' implicitly has an 'any' type.

2   function: ({ property }) => {
```

--- a/packages/nuxt/src/core/templates.ts
+++ b/packages/nuxt/src/core/templates.ts
@@ -399,6 +399,7 @@ export const nitroSchemaTemplate: NuxtTemplate = {
 
     return /* typescript */`
 ${lines.join('\n')}
+/// <reference path="./app.config.d.ts" />
 /// <reference path="./runtime-config.d.ts" />
 
 import type { RuntimeConfig } from 'nuxt/schema'

